### ISSUE TITLE
On restart during OldBodies goerli node crashed with "Attempted to process a blockchain without having starting state"

### ISSUE THREAD

**Issue by kamilchodola** [NethermindEth/nethermind#5996]
**Description**
On my node i have quite a few restarts (mixed graceful shutdowns and killed docker containers).
During one of the restarts on OldBodies stage i got an exception:
```
Processing loop threw an exception. Block: 9495485 (0x5a336a...b06939), Exception: System.InvalidOperationException: Attempted to process a blockchain without having starting state
   at Nethermind.Consensus.Processing.BlockchainProcessor.PrepareBlocksToProcess(Block suggestedBlock, ProcessingOptions options, ProcessingBranch processingBranch) in /src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs:line 565
   at Nethermind.Consensus.Processing.BlockchainProcessor.Process(Block suggestedBlock, ProcessingOptions options, IBlockTracer tracer) in /src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs:line 383
   at Nethermind.Consensus.Processing.BlockchainProcessor.RunProcessingLoop() in /src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs:line 316
```
After following restarts it did not get back to life - it just synced bodies and receipts further but block processing stopped.
Below are DEBUG logs from the moment it happened + CL Logs
[cl_logs.txt](https://github.com/NethermindEth/nethermind/files/12319976/cl_logs.txt)
[goerli_debug_logs_greped.zip](https://github.com/NethermindEth/nethermind/files/12319977/goerli_debug_logs_greped.zip)

**Desktop (please complete the following information):**
Please provide the following information regarding your setup:
 - Operating System: Ubuntu
 - Version: master 1.21.0-unstable
 - Installation Method: Sedge
 - Consensus Client: Teku

**Comment by kamilchodola** [comment]
It happened again now on gnosis chain.
Issue occurred again right after state sync on very beginning of OldBodies.

Debug logs for that:
[Gnosis_exception.zip](https://github.com/NethermindEth/nethermind/files/12416542/Gnosis_exception.zip)

**Comment by asdacap** [comment]
The gnosis one seems to store a different reorg boundary (off by one).
The goerli one to be setting head to an unexpected block, likely due to the block fixer. 
Its likely of different cause.

**Comment by asdacap** [comment]
For gnosis, the missing state is the last state root of statesync/healing. Its clearly downloaded.

**Comment by asdacap** [comment]
For goerli, blocktree initializer detected a corruption of some sort and attempted to set the head to the best suggested head. Unfortunately because of in memory pruning, the state for that block was not saved yet because of reorg depth.

```
2023-08-10 18:09:48.9876|DEBUG|19|Saving reorg boundary 9495427 

2023-08-10 18:10:24.6890|INFO|19|Processed             9495447     | 40,006.56 ms  |  slot     41,531 ms | Gas gwei: 0.14 .. 0.19 (2.69) .. 100.00 
2023-08-10 18:10:24.6890|INFO|19|- Block              12.87 MGas   |     68    txs |  calls  9,750 (143) | sload  35,982 | sstore  8,210 | create   7 
2023-08-10 18:10:24.6890|INFO|19|- Block throughput    0.32 MGas/s |      1.70 t/s |          0.00 Blk/s | recv        0 | proc       34 
2023-08-10 18:10:24.6890|DEBUG|19|- Total throughput    0.31 MGas/s |      1.64 t/s |          0.00 Blk/s | Gas gwei: 0.13 .. 0.15 (2.90) .. 100.00 
2023-08-10 18:11:16.8818|INFO|19|Processed     9495483...  9495484 |  4,800.23 ms  |  slot      8,779 ms | Gas gwei: 0.11 .. 0.11 (2.75) .. 100.00 
2023-08-10 18:11:16.8818|INFO|19|- Blocks 2           27.73 MGas   |    166    txs |  calls  1,119 ( 17) | sload   4,017 | sstore    932 | create   4 
2023-08-10 18:11:16.8818|INFO|19|- Block throughput    5.78 MGas/s |     34.58 t/s |          0.42 Blk/s | recv        0 | proc        0 
2023-08-10 18:11:16.8818|DEBUG|19|- Total throughput    6.10 MGas/s |     30.62 t/s |          0.41 Blk/s | Gas gwei: 0.10 .. 0.28 (2.85) .. 130.00 

2023-08-10 18:11:16.9907|DEBUG|61|Valid. Request: ForkChoice: Head: 9495484 (0xd51dc5...f1813f), Safe: 9495453 (0x97cf9c...9d1fca), Finalized: 9495425 (0xf3d257...a304bc). 

2023-08-10 18:11:22.2487|DEBUG|15|Executing step: InitializeBlockTree : InitializeBlockTree (WaitingForExecution) 
2023-08-10 18:11:22.3732|INFO|15|Start block loaded from reorg boundary - 9495427 - 9495427 (0x2611fc...ece740) 
2023-08-10 18:11:22.5170|INFO|15|Numbers resolved, level = Max(9495427, 9495484), header = Max(9495427, 9495484), body = Max(0, 9495484) 
2023-08-10 18:11:22.5780|INFO|15|Beacon Numbers resolved, level = 9495426, header = 0, body = 9495426 
2023-08-10 18:11:22.5780|WARN|15|Detected corrupted block tree data (0 < 9495426) (possibly due to an unexpected shutdown). Attempting to fix by moving head backwards. This may fail and you may need to resync the node. 
2023-08-10 18:11:22.5994|INFO|15|Block tree initialized, last processed is 9495484 (0xd51dc5...f1813f), best queued is 9495484, best known is 9495484, lowest inserted header 1, body 9469168, lowest sync inserted block number 9494583 
```
